File: 0020-valid-parentheses/0020-valid-parentheses.ts
function isValid(s: string): boolean {
    let len: number = s.length;
    let stack = [];
    let parenthesisMap = {
        '(': ')',
        '{': '}',
        '[': ']'
    };
    for (let i = 0; i < len; i++) {
        if (parenthesisMap[stack[stack.length - 1]] == s[i]) {
            stack.pop();
        } else {
            stack.push(s[i]);
        }
    }
    return stack.length == 0;
};

File: 0020-valid-parentheses/0020-valid-parentheses.java
import java.util.*;

class Solution {
    public boolean isValid(String s) {
        Map<Character, Character> map = Map.of(
            ')', '(',
            '}', '{',
            ']', '['
        );
        
        Stack<Character> stack = new Stack<>();
        
        for (char c : s.toCharArray()) {
            // If it's a closing bracket
            if (map.containsKey(c)) {
                if (stack.isEmpty() || stack.pop() != map.get(c)) {
                    return false;
                }
            }
            // If it's an opening bracket
            else if (map.containsValue(c)) {
                stack.push(c);
            }
            // Ignore everything else
        }
        
        return stack.isEmpty();
    }
}